face/UnifiedFaceCollection/python/unified_face_collection.py

```diff
@@ -175,223 +188,140 @@ def remove_person(self, person_identifier, delete_faces=False):
-        face_list_train_url = f"{self.face_api_url}/largefacelists/{self.large_face_list_id}/train"
-        face_list_train_response = requests.post(face_list_train_url, headers=self.headers)
-        
-        if face_list_train_response.status_code != 202:
-            print("Failed to initiate training for Large Face List")
-            return False
+        poller_face_list = self.face_admin_client.large_face_list.begin_train(
+            large_face_list_id=self.large_face_list_id,
+            polling_interval=5,
+        )
+        poller_face_list.wait()
+        print(f"The face list {self.large_face_list_id} is trained successfully.")
 
         # Train the Large Person Group
-        person_group_train_url = f"{self.face_api_url}/largepersongroups/{self.large_person_group_id}/train"
-        person_group_train_response = requests.post(person_group_train_url, headers=self.headers)
-        
-        if person_group_train_response.status_code != 202:
-            print("Failed to initiate training for Large Person Group")
-            return False
-
-        # Check training status for Large Face List
-        face_list_status_url = f"{self.face_api_url}/largefacelists/{self.large_face_list_id}/training"
-        while True:
-            face_list_status_response = requests.get(face_list_status_url, headers=self.headers)
-            face_list_status = face_list_status_response.json()
-            if face_list_status['status'] == 'succeeded':
-                break
-            elif face_list_status['status'] == 'failed':
-                print("Large Face List training failed")
-                return False
-            time.sleep(1)
-        
-        # Check training status for Large Person Group
-        person_group_status_url = f"{self.face_api_url}/largepersongroups/{self.large_person_group_id}/training"
-        while True:
-            person_group_status_response = requests.get(person_group_status_url, headers=self.headers)
-            person_group_status = person_group_status_response.json()
-            if person_group_status['status'] == 'succeeded':
-                break
-            elif person_group_status['status'] == 'failed':
-                print("Large Person Group training failed")
-                return False
-            time.sleep(1)
+        poller_person_group = self.face_admin_client.large_person_group.begin_train(
+            large_person_group_id=self.large_person_group_id,
+            polling_interval=5,
+        )
+        poller_person_group.wait()
+        print(f"The person group {self.large_person_group_id} is trained successfully.")
         
         return True
 
     def find_face(self, image_path, search_type='face'):
         # Detect face in the image
-        detected_faces = detect_faces(self.subscription_key, self.endpoint, image_path, self.injection_header)
+        detected_faces = self.detect_faces(image_path)
         face_ids = [face['faceId'] for face in detected_faces]
```

Comment:
```suggestion
        face_ids = [face.face_id for face in detected_faces]
```

See https://learn.microsoft.com/en-us/python/api/azure-ai-vision-face/azure.ai.vision.face.models.facedetectionresult?view=azure-python-preview#azure-ai-vision-face-models-facedetectionresult-face-id.